src/macrofier/types.mjs

```diff
@@ -614,11 +635,17 @@ function getSchemaType(schema, module, { destination, templateDir = 'types', lin
     // }
   }
   else if (schema.type) {
-    // TODO: this assumes that when type is an array of types, that it's one other primative & 'null', which isn't necessarily true.
-    const schemaType = !Array.isArray(schema.type) ? schema.type : schema.type.find(t => t !== 'null')
-    const primitive = getPrimitiveType(schemaType, templateDir)
-    const type = allocatedProxy ? allocatedPrimitiveProxies[schemaType] || primitive : primitive
-    return wrap(type, code ? '`' : '')
+    const template = getTemplate(path.join(templateDir, 'additionalProperties'))
```

Comment:
The current method (`getSchemaType`) doesn't recurse into each property... 

It looks like you're inserting the `Title` template if there happens to be a template for `additionalProperties`

I'm not sure I follow that. Could you explain?

Reply:
additionalPorperties always seening like a map. FlatMap is additional property and it is map. Here we have to get only its title. FlatMap. hence added logic to get Title.